### ISSUE TITLE
[Sept Sprint] New Exercise for Strings

### ISSUE THREAD

**Issue by iHiD** [exercism/go#1613]
This is part of our September Spring to get the Go Track in shape. Please read the [Overview issue](https://github.com/exercism/go/issues/1610) for more details. **This exercise is unclaimed. Please comment below if you'd like to work on it.**

---

**Before building this exercise, please read up on [the documentation](https://exercism.org/docs/building/tracks/concept-exercises).**

Copy/paste/modify from the [corresponding Javascript exercise](https://github.com/exercism/javascript/blob/main/exercises/concept/poetry-club-door-policy/). This exercise will be teaching the `strings` concept instead of having `logs-logs-logs` teach it. The latter will be modified to teach about `runes`. 

You can reuse lots of the text from here: https://github.com/exercism/go/blob/e41496d3ae8e03b2bf17ea0686c3fe78f1206ba9/exercises/concept/party-robot/.docs/introduction.md (That text is getting removed)

- The new exercise should give a basic introduction to strings (double quotes, escape sequences, backticks, `+`). 
-  Additionally this exercise should introduce the strings package. This can be done as part of the `strings` concept or as a separate concept `strings-package`. In any case this content should show how to access things from the package and show one usage example and a table with commonly used functions from the package. The concept about page should link to the full list of functions from the docs.
- It must  **not** use runes, slices, mutation, UTF-8 encoding or string formatting. This might require changes to the exercise.

The new exercise will have one prerequisite: `basics`.

For reference, the Swift track has also [ported the JS exercise](https://github.com/exercism/swift/tree/main/exercises/concept/poetry-club).

**Comment by junedev** [comment]
@ErikSchierboom I just realized the poetry club door policy exercise relies on syntax like `str[0]`. In Go that does not return a string and we didn't want to go into the details of runes, type conversion etc at this point because this can be the second learning exercise the student does on the Go track. Not sure how to "fix" that problem yet. Maybe we need to chat about it on Monday.

**Comment by bobtfish** [comment]
Text removed from the `string-formatting` concept in issue #1625 to be used in this concept / exercise is below:

# About

A `string` in Go is an immutable sequence of bytes. Strings may contain arbitrary bytes but usually they contain human-readable text.
Text strings are conventionally interpreted as UTF-8 encoded sequence of Unicode code points (runes) which will be explained in a future exercise.
A `string` value can be written as a string literal, which is a sequence of bytes enclosed in double quotes:

```go
s := "Hello World!"
```

The `+` operator makes a new string by [concatenating][string concatenation] two strings:

```go
fmt.Println("Hello" + " world!")
```

String values are immutable: the byte sequence contained in a string value can never be changed, though of course we can assign a new value to a string variable.
To append one string to another, we can use the `+=` statement:

```go
s := "Hello"
s += " word."
fmt.Println(s)
// Output: Hello world.
```

 # Introduction

A `string` in Go is an immutable sequence of bytes. Strings may contain arbitrary bytes but usually they contain human-readable text.
Text strings are conventionally interpreted as UTF-8 encoded sequence of Unicode code points (runes) which will be explained in a future exercise.
A `string` value can be written as a string literal, which is a sequence of bytes enclosed in double quotes:

```go
s := "Hello World!"
```

**Comment by junedev** [comment]
Instead of using poetry door club policy, I created a new story (+ tasks) that fits the things we can reasonably use in this exercise.
Note that I had to remove the code blocks so this renders properly.

```markdown
**Greeting Display**

There is an appliance store called "Tech Palace" nearby. The owner of the store  installed a big display that consists of lots of small LED lights. The display can show multiple lines of text and the store owner wants to use it to show a special greetings in case a customer scans their loyalty card at the entrance.

The store owner needs your help with the code that is used to generate the text for the new display.

1. Create the welcome message

For the normal customers, the store owner wants to see `Welcome to the Tech Palace, ` following by the name of the customer in capital letters on the display.

Implement the function `welcomeMessage` that accepts the name of the customer as string as an argument and returns the desired message.

welcomeMessage("Judy")
// Output: Welcome to the Tech Palace, JUDY

2. Add a fancy border
For loyal customers that buy a lot at the store, the owner wants the welcome display to be more fancy. They want a line of stars before and after the welcome text. The owner is not sure yet how many stars should be in the lines so they want that to be configurable.

Write a function `addBorder` that accepts a welcome text (a string) and the number of stars per line (type `int`) as arguments. It should return a string that consists of 3 lines, a line with the desired number of stars, then the welcome text as it was passed in, then another line of stars.

addBorder("Welcome!", 10)
// Output:
// **********
// Welcome!
// **********

Exemplar could look like this:
stars := strings.Repeat("*", num)
return stars + "\n" + text + "\n" + stars

3. Clean up old marketing messages
Some time ago the store already had a similar display but that one could only show some static lines that were set. From that time, there is a collection of  some nice marketing messages that could also be used for the new display. However, the data already includes a star border an some unfortunate whitespaces. Your task is to clean up the messages so they can be re-used.

Implement a function `cleanUpMessage` that accepts the old marketing message as a string. The function should first remove all stars from the text and afterwards remove the leading and trailing whitespaces from the remaining text. The function should then return the cleaned up message.

message := `
**************************
*    BUY NOW, SAVE 10%   *
**************************
`

cleanUpMessage(message)
// Output: BUY NOW, SAVE 10%

Exemplar could look like this:
msgWithoutStars := strings.ReplaceAll(message, "*", "")
return strings.TrimSpace(msgWithoutStars)

```

The first task is a bit similar to party robot but we want to make the party robot tasks a bit more advanced in the future anyway so I think this is ok for now.

**Comment by junedev** [comment]
Note: Once this exercise is in place, we need to check which `string-formatting` prerequisite entries can now be changed to `strings` again.

**Comment by ErikSchierboom** [comment]
I'll get the basics setup, and someone else can do the Go coding.

**Comment by ErikSchierboom** [comment]
> Note that I had to remove the code blocks so this renders properly.

Quick tip: you don't need to remove the code blocks if you make sure the outer code block uses a number of backticks different from the inner code blocks.

**Comment by junedev** [comment]
@ErikSchierboom Thanks for the tip! I didnt know that and didn't had time to search for a solution before posting it.

**Comment by ErikSchierboom** [comment]
@junedev Isn't `strings-package` supposed to be introduced in https://github.com/exercism/go/issues/1596?

**Comment by ErikSchierboom** [comment]
Ah, I see that we might want to change that exercise.

**Comment by junedev** [comment]
@ErikSchierboom Yes, sorry for the confusion, I had to sort this out over the weekend before people work into the wrong direction. The issue was that since strings don't have methods and the index does not already correspond to the letter, you can't do anything meaningful with them without the strings package. That is why I moved that part from logs,logs,logs to here.
